### ISSUE TITLE
Passing an enum type as a ``converter`` argument causes type check error

### ISSUE THREAD

**Issue by Fuyukai** [microsoft/pyright#6529]
**Describe the bug**
Pasing an enum's type as a ``converter`` argument for an ``attr.ib`` causes a type check error. Passing the ``__call__`` method directly (i.e. ``converter=MyEnum.__call__``) works correctly.

**Code or Screenshots**
```python
from enum import Enum
import attr

class TestEnum(Enum):
    ONE = 1
    TWO = 2

@attr.s()
class Test:
    field: TestEnum = attr.ib(converter=TestEnum)
```

Which gives the error:

```
/tmp/test/test.py
  /tmp/test/test.py:11:31 - error: Argument of type "(value: object) -> Self@Enum" is not a valid converter for field "field" of type "TestEnum"
    Type "(value: object) -> Self@Enum" cannot be assigned to type "(__input: __converterInput, /) -> TestEnum"
      Function return type "Self@Enum" is incompatible with type "TestEnum"
        "Enum" is incompatible with "TestEnum" (reportGeneralTypeIssues)
```

**VS Code extension or command-line**
Command line, 1.1.337 (nice version number).

**Comment by erictraut** [comment]
Type checking for the `attrs` library makes use of the `dataclass_transform` introduced in [PEP 681](https://peps.python.org/pep-0681/). This mechanism doesn't currently support `converter` because this isn't a feature of the stdlib `dataclass`. There is a draft [PEP 712](https://peps.python.org/pep-0712/) that proposes to add this support for both `dataclass` and `dataclass_transform`. Pyright contains provisional support for this draft PEP, but the PEP hasn't yet been approved by the Python steering council. There is a healthy debate going on [in this thread](https://discuss.python.org/t/pep-712-adding-a-converter-parameter-to-dataclasses-field) about the PEP. The SC has indicated that they are inclined to reject it. If they do, we will remove support for this entirely within pyright. If you value this feature, you should consider chiming in on that discussion.

The bug that you're reporting is due to pyright's provisional support for PEP 712. I agree it's a bug, and I'll work on a fix.

I'll note that if we do remove the provisional support, the error that you're seeing would no longer be reported, but you would lose all type checking for the converter parameter.

**Comment by erictraut** [comment]
This will be addressed in the next release of pyright and a future release of pylance.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.338, which I just published. It will also be included in a future release of pylance.
